### DISCUSSION TITLE
Next.js Blog Dynamic Routes Tutorial - 404 when visiting `/pages/[id]`

### DISCUSSION THREAD

**Discussion by ChromeUniverse** [vercel/next.js#45007] [Help] [answered]
Hi everyone,

I've started learning Next.js by following the [starter blog tutorial](https://nextjs.org/learn/basics) but I'm stuck in the ["Implement getStaticProps"](https://nextjs.org/learn/basics/dynamic-routes/implement-getstaticprops) section. I'm getting a 404 error whenever I try to visit `localhost:3000/posts/ssr-ssg` or `localhost:3000/posts/pre-rendering`. 

I've triple-checked that `pages/posts/[id].js` and `lib/posts.js` match the files in the [`next-learn`](https://github.com/vercel/next-learn/blob/master/basics/dynamic-routes-step-1/) repo, as suggested by the tutorial's troubleshooting section, but it's still not working. I'm not sure if it's an issue with `getStaticPaths()` or `getStaticProps()` in the `pages/posts/[id].js` file.

I've uploaded my files to a [public repo](https://github.com/ChromeUniverse/nextjs-blog). I would appreciate it if someone could clone it and take a look.

Thank you.

**Comment by scerelli** [comment, marked as answer]
I forked your repo, ran it and i don't get any 404. Are you sure you are typing the right url?

![Screenshot 2023-01-18 at 19 20 13](https://user-images.githubusercontent.com/4821190/213262734-fc74f177-0efd-48cc-8e94-1b489ef540f0.png)

**Reply by ChromeUniverse** [reply]
Yep, I was typing in the wrong URL, silly me. Thank you!
